core/src/test/java/com/graphhopper/routing/ch/PrepareContractionHierarchiesTest.java

```diff
@@ -363,6 +360,57 @@ public void testUnpackingOrder_Fastest() {
         assertEquals(IntArrayList.from(10, 0, 1, 2, 3, 4, 5, 6), p.calcNodes());
     }
 
+    @Test
+    public void testDisconnects() {
+        final GraphHopperStorage g = createGHStorage();
+        CHGraph lg = g.getGraph(CHGraph.class);
+        //            4
+        //            v
+        //            0
+        //            v
+        //  8 -> 3 -> 6 -> 1 -> 5
+        //            v
+        //            2
+        //            v
+        //            7
+        g.edge(8, 3, 1, false);
+        g.edge(3, 6, 1, false);
+        g.edge(6, 1, 1, false);
+        g.edge(1, 5, 1, false);
+        g.edge(4, 0, 1, false);
+        g.edge(0, 6, 1, false);
+        g.edge(6, 2, 1, false);
+        g.edge(2, 7, 1, false);
+        g.freeze();
+
+        PrepareContractionHierarchies prepare = createPrepareContractionHierarchies(g, lg)
+                .useFixedNodeOrdering(new NodeOrderingProvider() {
+                    @Override
+                    public int getNodeIdForLevel(int level) {
+                        return level;
+                    }
+
+                    @Override
+                    public int getNumNodes() {
+                        return g.getNodes();
+                    }
+                });
+        prepare.doWork();
+        CHEdgeExplorer explorer = lg.createEdgeExplorer();
+        // shortcuts leading to or coming from higher level nodes should be disconnected
```

Comment:
btw actually I thought this was quite interesting ;)

Reply:
You mean if we store the node ordering there could be some more edges than before this change?

Reply:
no like I said this is rather unrelated. I just pointed out here that we only disconnect shortcuts during CH preparation (because we can/do not change the base graph), but I think this definitely slows down the queries as we iterate many edges that lead to (or come from) lower level nodes anyway.

Reply:
This took me quite some time to realize.

Reply:
> (because we can/do not change the base graph),

As we have a special nodeCH DataAccess we store a different list of edges per node and we should be able to remove this per CH-profile. 

So this is the reason I was thinking the same: that we do this disconnect for base graph edges too.

> Sorry for the confusion I accidentally commited this test ;)

No problem. Maybe we should add this comment in the code instead of the test?

Reply:
> As we have a special nodeCH DataAccess we store a different list of edges per node and we should be able to remove this per CH-profile.

Good point, but are you sure this will work ? We only store the pointer to the first edge in nodeCH so not sure how this will work for edges that are further behind in the linked list of edges ? I was more thinking in the direction of storing the edge connections separately and reusing the rest of the data in base graph (flags etc.) by edge id, but seems to be a bigger issue.

> Maybe we should add this comment in the code instead of the test?

done

Reply:
Ah, yes. The linked list pointer (nodeCH) points into the shortcuts first and then the last entry of the shortcuts points into the original edges. So yes, duplicating this information would be a bigger issue and increase RAM usage further (but maybe faster as you said).